Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -32,30 +32,88 @@ namespace ROS2
 {
     URDFPrefabMaker::URDFPrefabMaker(
         const AZStd::string& modelFilePath,
-        urdf::ModelInterfaceSharedPtr model,
+        const sdf::Root* root,
         AZStd::string prefabPath,
         const AZStd::shared_ptr<Utils::UrdfAssetMap> urdfAssetsMapping,
         bool useArticulations,
         const AZStd::optional<AZ::Transform> spawnPosition)
-        : m_model(model)
-        , m_visualsMaker(model->materials_, urdfAssetsMapping)
+        : m_root(root)
+        , m_visualsMaker{}
         , m_collidersMaker(urdfAssetsMapping)
         , m_prefabPath(std::move(prefabPath))
         , m_urdfAssetsMapping(urdfAssetsMapping)
         , m_spawnPosition(spawnPosition)
         , m_useArticulations(useArticulations)
     {
         AZ_Assert(!m_prefabPath.empty(), "Prefab path is empty");
-        AZ_Assert(m_model, "Model is nullptr");
+        AZ_Assert(m_root, "SDF Root is nullptr");
+        if (m_root != nullptr)
+        {
+            AZ_Assert(GetFirstModel(), "SDF Model is nullptr");
+
+            AZStd::unordered_map<AZStd::string, const sdf::Material*> materialMap;
```

Comment:
Suggestion: might want to define a `using MaterialNameMap = AZStd::unordered_map<AZStd::string, const sdf::Material*>;` somewhere.

Reply:
Will do